### PR TITLE
Exclude SemanticNodes that are fully covered by an overlapping sliver

### PR DESCRIPTION
**Pull request by goderbauer**
There are two cases to consider:
* The entire sliver is fully covered by another overlapping sliver.
* Only parts of a sliver are overlapped by a sliver and those parts fully cover some SemanticsNodes.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart [hunk 1]
```diff
@@ -285,6 +285,34 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
       visitor(child);
   }
 
+  @override
+  void visitChildrenForSemantics(RenderObjectVisitor visitor) {
+    switch(constraints.normalizedGrowthDirection) {
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart]: nit: space before (

File: packages/flutter/test/widgets/sliver_semantics_test.dart [hunk 2]
```diff
@@ -281,4 +282,326 @@ void main() {
...
+                tags: <SemanticsTag>[RenderSemanticsGestureHandler.excludeFromScrolling],
+              ),
+            ],
+          )
+        ],
+      ),
+    ));
+
+    semantics.dispose();
+  });
```
Comment (Hixie) [inline packages/flutter/test/widgets/sliver_semantics_test.dart]: test with a center: widget and positive and negative overlapped slivers?

**Comment by Hixie** [comment]
LGTM

**Comment by goderbauer** [comment]
Added tests for center widget and fixed nits. Thanks!

**Pull request merged**